server/utils/cloudFunctions/functions/test/algoliaTests/algoliaIntegrationTests.test.js

```diff
@@ -0,0 +1,237 @@
+ *   `getObjects`. 
+ *   - `getObject` takes a string `objectID` returns a json object. 
+ *     - If the object doesn't exist, an error will be thrown. 
+ *   - `getObjects` takes a string list of `objectID`, and returns a json 
+ *     object. 
+ *     - If all records exist, the document is of the form 
+ *       `{ results: [record1, record2, ...]}`
+ *     - If a record doesn't exist, its index in the list will be `null`.
+ *       There will also be an additional field `message`:
+ *       `{ results, message }`. The message is of the form `'ObjectID {records objectID} does not exist. '`
+ *     - Because `getObjects` does not throw an error, it is more useful than 
+ *       `getObject` to validate a record has been deleted - if the query 
+ *       fails/throws an error for a different reason than the `getObject` 
+ *       error, the test may still pass. Checking for a `null` field
+ *       and a string equality match is alternatively much easier. 
+ *     - For the above reason, `getObjects` is used with singleton lists
+ *       to verify that a record has been deleted instead of `getObject`. 
+ * - `setTimeout` is used to allow firebase changes to propogate to algolia. 
+ *   - `setTimeout` requires that `done()` be called at the end of the 
+ *     inner function body. It is not equivalent to `return`. 
+ *   - Due to the use of `done()` at the end of `setTimeout`, promises
+ *     are not returned as they are in `countAdvertisersFunction` test suite.
+ *   - This is because it will make the test `overspecified`
+ *   - Read more here: `https://medium.com/@durja/mocha-tests-with-async-await-2aead4afeca1`
+ *   - This does not impact testing at all, but is a notable style difference.
+ * - `TIMEOUT_10S` was chosen because this is the maximum amount of latency 
+ *   for firebase to execute a function. 
+ * - `TIMEOUT_15S` is used to extend the default 2000ms test limit set by 
+ *   mocha.
+ * - The overall test timeout limit must be longer than the `setTimeout`
+ *   method to allow time for the algolia api call.    
+ * Tests are structured with promise nesting. 
+ * - There is a debate about nesting vs chaining promises. In this case, 
+ *     nesting is advantageous because it easily keeps everything in scope
+ *     without much fuss. Thus, nesting is fine to use. 
+ * - Also this is how the official docs do it. 
+ * - `Promise.allSettled` is used to avoid nesting hell where possible. 
+ *   Sometimes not possible if their is some sequential order that
+ *   must be obeyed (e.g., create and then delete.)  
+ * - Looked into using `await/async` but ran into more flakiness for some
+ *   reason. 
+ * Errors are handled directly by firebase, so their is no need to validate
+ * how errors are handled/resolved.   
+ * Author: Robert Marcus
+ * Date: July 16, 2020
+ */
+
+
+// Import the testing environment configuration.
+const { chai,
+        TIMEOUT_10S,
+        TIMEOUT_15S,
```

Comment:
Why 10 and 15 seconds? Will this ever change?

Reply:
There is a comment about this above and where these constants are declared, but I added another comment here so one doesn't have to search for context.